src/builder/yara_rule_builder.cpp

```diff
@@ -85,7 +122,42 @@ YaraRuleBuilder& YaraRuleBuilder::withModifier(Rule::Modifier mod)
  */
 YaraRuleBuilder& YaraRuleBuilder::withTag(const std::string& tag)
 {
-	_tags.push_back(tag);
+	if (tag == "")
+		throw RuleBuilderError("Error: tag must be non-empty.");
+	if (_tags.empty())
+		_tokenStream->emplace(_lcb, COLON, ":");
+	TokenIt it = _tokenStream->emplace(_lcb, TokenType::TAG, tag);
+	_tags.push_back(it);
+	return *this;
+}
+
+/**
+ * Adds comment to a rule.
+ *
+ * @param comment Comment.
+ *
+ * @return Builder.
+ */
+YaraRuleBuilder& YaraRuleBuilder::withComment(const std::string& comment, bool multiline)
+{
+	if (comment == "")
+		throw RuleBuilderError("Error: comment must be non-empty.");
+	TokenIt insert_before = _mod.value_or(_rule_it);
+	std::stringstream ss;
+	if (multiline)
+	{
+		ss << "/* " << comment << " */";
+		_tokenStream->emplace(insert_before, TokenType::COMMENT, ss.str());
+	}
+	else
+	{
+		for (auto item : comment)
```

Comment:
There is already `find` method (https://en.cppreference.com/w/cpp/string/basic_string/find)